### ISSUE TITLE
docker base

### ISSUE THREAD

**Issue by nsheff** [databio/LOLAweb#45]
instead of starting from the R base image in the docker file, we should consider a few alternatives:

- using a bioconductor container?
- creating our own image and using that as the base image?

As it is, the way I 'm seeing this, it's is a 25 minute travis build for deploy because it's re-installing lots of stuff from scratch every time. We could avoid this by having the LOLA base image start from something a little more refined.

**Comment by nsheff** [comment]
Further:

I think the app itself should actually not be pulled into the dockerfiles.

I have this set up now locally where I have a container (`nsheff/rdev` to be precise) that can host my app, which I just run using 

```
docker exec -it nsheff/rdev R -e "shiny::runApp('${package}',host='0.0.0.0',port=4020,launch.browser=FALSE)"
```

This decouples the docker container from the app; changes to the app need only update the app. Changes to the container should be the only reason to rebuild/update the container. Right now these are coupled.

**Comment by nsheff** [comment]
BTW, it dropped to an 8 minute (ish) build when I removed the BSgenome dependencies.

**Comment by nmagee** [comment]
Okay, so we are ready to move to a new Docker base image.

Code:
https://github.com/databio/shinyBase/

Docker image:
https://hub.docker.com/r/databio/shinybase/

Travis builds:
https://travis-ci.org/databio/shinyBase

So in LOLAweb Dockerfiles we can now remove lines 5-29 and just use a new base:

```
FROM databio/shinybase:latest
```

New builds of LOLAweb should take only a few minutes.

**Comment by nmagee** [comment]
Build time for the new base container is ~ 8 min

**Comment by nmagee** [comment]
w0w

Build time with the new base image was 1 min 18 sec.

**Comment by nsheff** [comment]
looking good -- everything seems to be working from here

**Comment by nmagee** [comment]
Yeah, it's really just a simple switchout -- it was a good suggestion.
